app/controllers/schemes_controller.rb

```diff
@@ -51,17 +51,34 @@ def new_deactivation
   end
 
   def deactivate_confirm
-    @affected_logs = @scheme.lettings_logs.visible.after_date(params[:deactivation_date])
-    if @affected_logs.count.zero?
+    @deactivation_date = Time.zone.parse(params[:deactivation_date])
+    @affected_logs = @scheme.lettings_logs.visible.after_date(@deactivation_date)
+    @deactivation_date_type = params[:deactivation_date_type]
+
+    scheme_locations = @scheme.locations.confirmed
+
+    locations_active_on_deactivation_date, remaining_locations = scheme_locations.partition do |location|
+      location.status_at(@deactivation_date) == :active
```

Comment:
What if the location is deactivating soon on deactivation date? Wouldn't that mean that it's technically still active?

Reply:
Actually it would make more sense for the deactivating soon to be in this partition. However, since we no longer need to display each of these three scenarios on the next page, I will put them all together into a single select.